src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -2085,6 +2085,11 @@ public override BoundStatement BindStatement(StatementSyntax node, DiagnosticBag
                 return (BoundStatement)boundNode;
             }
 
+            internal override BoundBlock BindEmbeddedBlock(BlockSyntax node, DiagnosticBag diagnostics)
```

Comment:
Why are we only having this issue with embedded blocks?

Reply:
Because everything else detours through `BindStatement`, but a few things with embedded blocks such as `try` and `catch` instead use this, which doesn't hook it. We have asserts in the `NodeMapBuilder` that assert we are always reusing statements, which this was hitting.

---
In reply to: [227889882](https://github.com/dotnet/roslyn/pull/30721#discussion_r227889882) [](ancestors = 227889882)